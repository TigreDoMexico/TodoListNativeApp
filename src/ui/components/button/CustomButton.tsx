import React from 'react';
import { TouchableOpacity, View, StyleSheet, ViewStyle } from 'react-native';

type Props = {
    onPress: (event: any) => void,
    children: React.ReactNode,
    customStyle?: ViewStyle;
    disabled?: boolean
}

const CustomButton = (props: Props) => {
    const isDisabled = props.disabled ? props.disabled : false;
    const onPressFunc = isDisabled ? () => {} : props.onPress;

    return(
        <TouchableOpacity testID='touchable_custom_button' disabled={isDisabled} onPress={onPressFunc}>
            <View style={[styles.button, props.customStyle]}>
                {props.children}
            </View>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    button: {
        alignItems: 'center',
        justifyContent: 'center',
    },
})

export default CustomButton;