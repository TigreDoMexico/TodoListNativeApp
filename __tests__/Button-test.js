import React from 'react';
import {Text} from 'react-native'
import { act, create } from 'react-test-renderer'
import CustomButton from '../src/ui/components/button/CustomButton'

const pressFunc = jest.fn();
const child = (<Text>Teste</Text>)
const buttonComponent = create(<CustomButton onPress={pressFunc}>{child}</CustomButton>)

test('Criar um CustomButton - Esperado combinar com o snapshot', () => {
    expect(buttonComponent).toMatchSnapshot();
});

test('Clicar no CustomButton - Esperado chamar a função pressFunc 1 vezes', () => {
    //Press Button
    const button = buttonComponent.root.findByProps({testID: 'touchable_custom_button'}).props;
    act(() => button.onPress())

    expect(pressFunc).toHaveBeenCalledTimes(1);
});