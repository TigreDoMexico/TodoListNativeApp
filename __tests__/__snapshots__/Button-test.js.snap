// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Criar um CustomButton - Esperado combinar com o snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="touchable_custom_button"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <Text>
      Teste
    </Text>
  </View>
</View>
`;
